### PR TITLE
Limit zipkin tag value length

### PR DESCRIPTION
**Pull request by pjanotti**
Implement tag truncation per current implementation: truncate tag values at export time.

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/TracerSettings.cs [hunk 1]
```diff
@@ -33,6 +33,8 @@ public class TracerSettings
         /// </summary>
         public const int DefaultAgentPort = 8126;
 
+        private const int DefaultRecordedValueMaxLength = 1200;
```
Comment (pellared) [inline tracer/src/Datadog.Trace/Configuration/TracerSettings.cs:36]: The default should be `12000`.
See: https://github.com/signalfx/gdi-specification/pull/130

File: tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs [hunk 2]
```diff
@@ -392,6 +392,14 @@ public static class ConfigurationKeys
         /// <seealso cref="TracerSettings.KafkaCreateConsumerScopeEnabled"/>
         public const string KafkaCreateConsumerScopeEnabled = "SIGNALFX_TRACE_KAFKA_CREATE_CONSUMER_SCOPE_ENABLED";
 
+        /// <summary>
+        /// Configuration key to set maximum length a tag/log value can have.
+        /// Values are completely truncated when set to 0, and ignored when set to negative
+        /// or non-integer string. The default value is 1200.
+        /// </summary>
+        /// <seealso cref="TracerSettings.RecordedValueMaxLength"/>
+        public const string RecordedValueMaxLength = "SIGNALFX_RECORDED_VALUE_MAX_LENGTH";
```
Comment (pellared) [inline tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs:401]: Can you document it in `USAGE.md`?

File: tracer/src/Datadog.Trace/Configuration/TracerSettings.cs (hunk 1 above)
Comment (pellared) [reply tracer/src/Datadog.Trace/Configuration/TracerSettings.cs]: Addressed 35138ef6e

File: tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs (hunk 2 above)
Comment (pellared) [reply tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs:401]: Addressed bcc865a3c1a8

**Review by pellared — APPROVED** [review]

**Review by RassK — APPROVED** [review]
LGTM

**Pull request merged**